src/main/scala/rise/elevate/rules/algorithmic.scala

```diff
@@ -28,11 +28,11 @@ object algorithmic {
 
   def  splitJoin(n: Nat): Strategy[Rise] = `*f -> S >> **f >> J`(n: Nat)
   @rule def `*f -> S >> **f >> J`(n: Nat): Strategy[Rise] = {
-    case e @ App(map(), f) => Success((split(n) >> map(map(f)) >> join) :: e.t)
+    case e @ App(map(), f) => Success((split(n) >> map(map(f)) >> join) !: e.t)
```

Comment:
this is not related to this PR, but maybe we could use a wrapper of partial function here, as in
```
@rule def `*f -> S >> **f >> J`(n: Nat): Strategy[Rise] = rewrite {
    case e @ App(map(), f) => split(n) >> map(map(f)) >> join
}
```
where rewrite wraps a `Success/Failure` and also adds the type assertion.

Reply:
Yes, we will want to do something like this.